python/paddle/v2/fluid/tests/test_target_location_assign_op.py

```diff
@@ -0,0 +1,167 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserve.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import numpy as np
+import math
+import sys
+import random
+from op_test import OpTest
+
+
+class TestTargetLocationAssginOp(OpTest):
+    def set_data(self):
+        self.init_test_case()
+
+        self.inputs = {
+            'Loc': self.loc_predic,
+            'GTBoxes': (self.encode_gt_boxes, self.encode_gt_boxes_lod),
+            'MatchIndices': self.match_indices,
+            'PriorBoxes': self.prior_boxes,
+            'PriorVariances': self.prior_variance
+        }
+
+        self.attrs = {
+            'encode_variance_in_target': self.encode_variance_in_target
+        }
+
+        self.outputs = {
+            'LocGT': self.loc_gt_data,
+            'LocPred': self.loc_pred_data
+        }
+
+    def init_test_case(self):
+        self.batch_size = 10
+        self.prior_num = 10
+        self.encode_gt_boxes_num = 100
+        self.encode_gt_boxes_lod = [[0, 5, 15, 30, 40, 50, 60, 70, 80, 90, 100]]
+        self.encode_variance_in_target = False
+
+        self.init_input_data()
+        self.loc_gt_data, self.loc_pred_data = self.calc_location_assign()
+
+    def init_input_data(self):
+        def make_box(init=0.0):
+            xmin = random.uniform(init, 1.0)
+            ymin = random.uniform(init, 1.0)
+            xmax = random.uniform(xmin, 1.0)
+            ymax = random.uniform(ymin, 1.0)
+            return [xmin, ymin, xmax, ymax]
+
+        # [batch_size, prior_num, 4]
+        self.loc_predic = np.zeros(
+            (self.batch_size, self.prior_num, 4)).astype('float32')
+
+        # [encode_gt_boxes_num, prior_num, 4]
+        self.encode_gt_boxes = np.zeros(
+            (self.encode_gt_boxes_num, self.prior_num, 4)).astype('float32')
+
+        # match_indices[n, p] = gt_box_index
+        self.match_indices = np.zeros(
+            (self.batch_size, self.prior_num)).astype('int32')
+
+        for n in range(self.batch_size):
+            gt_start = self.encode_gt_boxes_lod[0][n]
+            gt_end = self.encode_gt_boxes_lod[0][n + 1]
+            gt_num = gt_end - gt_start
+            for p in range(self.prior_num):
+                self.loc_predic[n, p, :] = make_box()
```

Comment:
In fact, `loc_predic` is the localization prediction offsets, not bbox. So using `random.random` is ok.